routes/conversation.js

```diff
@@ -255,6 +257,74 @@ exports.leave = function (req, res){
 	});
 };
 
+/**
+ * Action when a user archives a conversation.
+ */
+exports.archive = function (req, res){
+	if (!req.user || !req.user._id){
+		resError(res, "Access denied.", "/error");
+		return;
+	}
+
+	// Search through the user's conversation list until the conversation with the correct
+	// id is found. 
+	User.findById( req.user._id, function (err, user){
+		if (err){
+			resError(res, "Could not find the user.");
+			return;
+		}
+
+		// Save the id of the conversation that we want to archive.
+		var targetConversationId = JSON.stringify(req.body.conversationId);
+
+		// Find the index at which the target conversation exists amongst all of the user's conversations.
+		var conversationIds = user.userConversations.map(function(p) {return p.conversation});
+		var conversationIdStrings = conversationIds.map(function(id) {return JSON.stringify(id)});
+		var foundIndex = conversationIdStrings.indexOf(targetConversationId);
+
+		// Modify the hallOfFame boolean associated with the conversation to true if it exists.
+		if (foundIndex !== -1) {
+			user.userConversations[foundIndex].hallOfFame = true;
+
+			// Save the modified user object.
+			user.save(function(err){
+				if (err) return resError(res, "Could not save user to database.");
+
+				// If there are no errors, proceed to update the isThrilled boolean of the user
+				// in the conversation object to true and save this object.
+				Conversation.findById(targetConversationId, function(err, convo){
+					if (err) return resError(res, "Could not find conversation");
+
+					// Save the id of the user whose isThrilled boolean we want to set as true.
+					var userString = JSON.stringify(req.user._id);
+
+					// Find the index at which the target user exists amongst all of the conversation's participants.
+					var participantIds = convo.participants.map(function (p){return p._id});
+					var participantIdStrings = participantIds.map(function (id){return JSON.stringify(id)});
+					var userIndex = participantIdStrings.indexOf(userString);
+
+					// Modify the isThrilled boolean associated with the appropriate participant if 
+					// he/she exists.
+					if (userIndex !== -1){
+						// Update the user's isThrilled boolean.
+						convo.participants[userIndex].isThrilled = true;
+
+						// Save the modified conversation.
+						convo.save(function(err){
+							if (err) return resError(res, "Could not save conversation to database.");
+
+							// After everything is done, send the response back to the user with success.
+							res.send({success: true, redirect: '/home'});
+						});
+					} else{
+						reject(res, "Access denied.", "/error");
+					}
+				});
```

Comment:
After pulling from the branch and running it locally, I got an error around this line, so it may be that one of the brackets/parens isn't matching up.  I think there might need to be another `});` after this line here.

Hopefully we find a better way of coping with all these callbacks soon :p   There's a neat node module called "async", which looks promising, but it might be better to do a callback refactoring in the future, as I know that I've written a lot of callbacks in places that I shouldn't have!